mlir/lib/Dialect/Tensor/Transforms/ReshapePatterns.cpp

```diff
@@ -79,12 +79,42 @@ struct FoldInsertOfRankReducingInsert : public OpRewritePattern<OpTy> {
     return success();
   }
 };
+
+/// Fold rank increasing expand_shape into insert_slice.
```

Comment:
`expand_shape` is always rank-increasing, so the comment is a bit ambiguous.

Reply:
Right, I couldn't settle on a nicer name.
Renamed to "padding" as it should reflect this type of expand better.